src/GitHub.App/ViewModels/PullRequestDetailViewModel.cs

```diff
@@ -411,6 +415,30 @@ string GetBranchDisplayName(string targetBranchLabel)
             }
         }
 
+        string GetStatusDisplay(IPullRequestFileModel file, TreeChanges changes)
+        {
+            switch (file.Status)
+            {
+                case PullRequestFileStatus.Added:
+                    return "add";
+                case PullRequestFileStatus.Renamed:
+                    var fileName = file.FileName.Replace("/", "\\");
+                    var change = changes?.Renamed.FirstOrDefault(x => x.Path == fileName);
+
+                    if (change != null)
+                    {
+                        return Path.GetDirectoryName(change.OldPath) == Path.GetDirectoryName(change.Path) ?
+                            Path.GetFileName(change.OldPath) : change.OldPath;
+                    }
+                    else
+                    {
+                        return "rename";
+                    }
+                default:
+                    return null;
```

Comment:
Also note that this property is likely only to be used for binding, so nulls shouldn't be a problem (famous last words I know...)